src/criteria_assessment/site_identification.jl

```diff
@@ -0,0 +1,262 @@
+
+# Arguments
+- `reg_assess_data` : Dictionary containing the regional data paths, reef outlines and full region names.
+- `reg` : Name of the region being assessed (format `Cairns-Cooktown` rather than `Cairns/Cooktown Management Area`).
+- `criteria_qp` : Dict containing bounds for each variable being filtered.
+- `assessment_qp` : Dict containing the dimensions of the search polygon (`xdist`, `ydist`) and the `SuitabilityThreshold` to identify search pixels.
+- `rtype` : Type of zone to assess (flats or slopes).
+- `config` : Information from `.config.toml` file.
+
+# Returns
+GeoJSON file containing result site polygons after filtering out polygons < 0.33 score and
+keeping the highest scoring polygon where intersecting polygons occur.
+"""
+function site_assess_region(reg_assess_data, reg, criteria_qp, assessment_qp, rtype, config)
+    @debug "Assessing region's suitability score"
+
+    file_id = string(hash(assessment_qp))
+    assessed_tmp_path = _cache_location(config)
+    assessed_path_geojson = joinpath(
+        assessed_tmp_path, file_id * "_potential_sites.geojson"
+    )
+
+    if isfile(assessed_path_geojson)
+        return file(assessed_path_geojson)
+    end
+
+    file_id = string(hash(criteria_qp))
+    assessed_tmp_path = _cache_location(config)
+    assessed_path_tif = joinpath(assessed_tmp_path, file_id * "_suitable.tiff")
+
+    if isfile(assessed_path_tif)
+        scan_locs = Raster(assessed_path_tif)
+    else
+        # Make mask of suitable locations
+        mask_data = _temp_assess_region(reg_assess_data, reg, criteria_qp, rtype, config)
+
+        # Assess remaining pixels for their suitability
+        @debug "Calculating proportional suitability score"
+        suitability_scores = proportion_suitable(mask_data.data)
+
+        # Need rebuild(mask_data, suitability_scores) as input to identify_potential_sites_edges
+        scan_locs = rebuild(mask_data, suitability_scores)
+    end
+
+    # Need dataframe of valid_lookup pixels
+    @debug "Pre-processing assessment inputs."
+    #Main.@infiltrate
+    crit_pixels = _temp_filter_lookup(reg_assess_data, reg, criteria_qp, rtype, config)
+
+    res = abs(step(dims(scan_locs, X)))
+    target_crs = convert(EPSG, crs(scan_locs))
+
+    suitability_threshold = parse(Int64, (assessment_qp["SuitabilityThreshold"]))
+    scan_locs = identify_search_pixels(scan_locs, x -> x .> suitability_threshold)
+
+    # Need reef outlines
+    gdf = REGIONAL_DATA["reef_outlines"]
+    reef_outlines = buffer_simplify(gdf)
+    reef_outlines = polygon_to_lines.(reef_outlines)
+
+    x_dist = parse(Int64, assessment_qp["xdist"])
+    y_dist = parse(Int64, assessment_qp["ydist"])
+
+    @debug "Performing polygon site assessment."
+    initial_polygons = identify_potential_sites_edges(
+        crit_pixels,
+        scan_locs,
+        res,
+        gdf,
+        x_dist,
+        y_dist,
+        target_crs,
+        reef_outlines,
+        REGIONAL_DATA["region_long_names"][reg]
```

Comment:
functions should always use the standard region id. Move `REGIONAL_DATA["region_long_names"][reg]` into the function.